### PR TITLE
feat: sync polaris-controller to polaris server.

### LINKED ISSUE #108: controller能够将自己作为一个服务注册到北极星中去
**What is the feature you want to add?**

controller能够将自己作为一个服务注册到北极星中去，通过 ProviderAPI.RegisterInstance 方法实现

**Why do you want to add this feature?**

用户可以直接从北极星的服务中查看 controller 的状态，命名空间默认在 Polaris，服务名称格式为 polaris-controller.{clusterName}

**How to implement this feature?**

**Additional context**

### PR DESCRIPTION
**Pull request by lhiamgeek**
**Please provide issue(s) of this PR:**
Fixes #108 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Docs
- [ ] Inject Sidecar
- [ ] Installation
- [X] Performance and Scalability
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/util/helper.go [hunk 1]
```diff
@@ -39,6 +39,10 @@ var ignoredNamespaces = []string{
 	"kube-node-lease",
 }
 
+var controllerNamespace = "polaris-system"
```
Comment (chuntaojun) [inline pkg/util/helper.go:42]: var (
   ignoreSyncService = map[string]struct{}{
      "service": {}
   }
)

File: pkg/util/helper.go [hunk 2]
```diff
@@ -143,6 +147,13 @@ func IsPolarisService(svc *v1.Service, syncMode string) bool {
 
 // IgnoreService 添加 service 时，忽略一些不需要处理的 service
 func IgnoreService(svc *v1.Service) bool {
+	// sync polaris-controller
+	if svc.Namespace == controllerNamespace && svc.Name == controllerServiceName {
```
Comment (chuntaojun) [inline pkg/util/helper.go:151]: 这里直接根据上面的 map 来做判断处理

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -1036,3 +1038,32 @@ func (p *PolarisController) checkHealth() {
...
+		req := &api.InstanceRegisterRequest{}
+		req.Service = "polaris-controller." + endpoint.ClusterName
+		req.ServiceToken = globalToken
+		req.Namespace = "Polaris"
+		req.Host = host
+		req.Port = int(port)
+		req.Healthy = healthy
+		req.SetTTL(5)
+
+		_, err := p.provider.Register(req)
```
Comment (chuntaojun) [inline pkg/controller/controller.go:1064]: 可以更新了 polaris-go 的版本，用 RegisterInstance

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -360,6 +360,8 @@ func (p *PolarisController) onEndpointAdd(obj interface{}) {
 	// 监听到创建service，处理是通常endpoint还没有创建，导致这时候读取endpoint 为not found。
 	endpoint := obj.(*v1.Endpoints)
 
+	p.registerController(endpoint)
```
Comment (chuntaojun) [inline pkg/controller/controller.go:363]: 挪到 controller 启动的时候处理吧，POD_IP 是可以通过注入到 ENV 拿到的

**Review by chuntaojun — CHANGES_REQUESTED** [review]

File: cmd/polaris-controller/app/polaris-controller-manager.go [hunk 5]
```diff
@@ -480,20 +484,61 @@ func ResyncPeriod(c *options.CompletedConfig) func() time.Duration {
 	}
 }
 
+// buildAPI
+func buildAPI() (consumerAPI api.ConsumerAPI, providerAPI api.ProviderAPI, err error) {
```
Comment (chuntaojun) [inline cmd/polaris-controller/app/polaris-controller-manager.go:488]: 不建议返回值带上名称，这里调整下吧

**Review by chuntaojun — CHANGES_REQUESTED** [review]

File: cmd/polaris-controller/app/polaris-controller-manager.go (hunk 5 above)
Comment (lhiamgeek) [reply cmd/polaris-controller/app/polaris-controller-manager.go:488]: ok

File: cmd/polaris-controller/app/polaris-controller-manager.go [hunk 6]
```diff
@@ -557,3 +602,43 @@ func readConfFromFile() (*controllerConfig, error) {
 //	).Run(ctx.ComponentConfig.PolarisController.ConcurrentPolarisSyncs, ctx.Stop)
 //	return nil, nil
 //}
+
+// registerController register controller to server
+func registerController(c *options.KubeControllerManagerConfiguration, provider api.ProviderAPI) {
+	req := &api.InstanceRegisterRequest{}
+	req.Service = "polaris-controller." + c.PolarisController.ClusterName
+	req.ServiceToken = globalToken
```
Comment (chuntaojun) [inline cmd/polaris-controller/app/polaris-controller-manager.go:610]: 这里不是用 globalToken，这里是北极星 用户/用户组 token

File: cmd/polaris-controller/app/polaris-controller-manager.go [hunk 7]
```diff
@@ -64,6 +66,7 @@ const (
 	PolarisAccountingControllerName = "polaris-accounting-controller"
 	DefaultLockObjectName           = "polaris-controller"
 	DefaultLeaderElectionName       = "polaris-controller"
+	globalToken                     = "polaris@12345678"
```
Comment (chuntaojun) [inline cmd/polaris-controller/app/polaris-controller-manager.go:69]: remove it

File: cmd/polaris-controller/app/polaris-controller-manager.go (hunk 6 above)
Comment (chuntaojun) [reply cmd/polaris-controller/app/polaris-controller-manager.go:610]: serviceSync:
      mode: "all"
      serverAddress: #POLARIS_HOST#
      # 北极星开启鉴权时需要配置
      accessToken: #POLARIS_TOKEN#

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**